From 1f0284d40f619af73dbb3dbb52694ca07d2fbe5b Mon Sep 17 00:00:00 2001
From: Harshita Goswami <harshita.goswami@intel.com>
Date: Wed, 4 Oct 2017 01:46:33 +0530
Subject: [PATCH] Code Changes for WIFI MAC address display

"Adding API for Display of WIFI MAC address in
Android Menu in Supplicant under INTEL_SUPPL_IWL_EXT flag"

Tracked-On: None

Signed-off-by: Harshita Goswami <harshita.goswami@intel.com>
---
 src/drivers/driver_nl80211.c         | 20 ++++++++++++++++++++
 src/drivers/driver_nl80211_android.c |  4 ++--
 wpa_supplicant/Android.mk            |  6 ++++++
 3 files changed, 28 insertions(+), 2 deletions(-)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index 2ce03ed..1d4dbf2 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -2396,6 +2396,26 @@ static int qca_vendor_test_cmd_handler(struct nl_msg *msg, void *arg)
 }
 #endif /* CONFIG_TESTING_OPTIONS */
 
+#ifdef INTEL_SUPPL_IWL_EXT
+int wpa_driver_nl80211_driver_cmd(void *priv, char *cmd, char *buf, size_t buf_len)
+{
+	struct i802_bss *bss = priv;
+	struct wpa_driver_nl80211_data *drv = bss->drv;
+	int ret = 0;
+	u8 macaddr[ETH_ALEN] = {};
+
+	if (os_strcasecmp(cmd, "MACADDR") == 0) {
+		ret = linux_get_ifhwaddr(drv->global->ioctl_sock, bss->ifname, macaddr);
+		if (!ret) {
+			ret = os_snprintf(buf, buf_len,"Macaddr = " MACSTR "\n", MAC2STR(macaddr));
+		}
+	} else {
+		wpa_printf(MSG_ERROR, "Unsupported command: %s", cmd);
+		ret = -1;
+	}
+	return ret;
+}
+#endif /* INTEL_SUPPL_IWL_EXT */
 
 static void qca_vendor_test(struct wpa_driver_nl80211_data *drv)
 {
diff --git a/src/drivers/driver_nl80211_android.c b/src/drivers/driver_nl80211_android.c
index ba47888..85521fd 100644
--- a/src/drivers/driver_nl80211_android.c
+++ b/src/drivers/driver_nl80211_android.c
@@ -151,7 +151,7 @@ int android_pno_stop(struct i802_bss *bss)
 
 
 #ifdef ANDROID_P2P
-#ifdef ANDROID_LIB_STUB
+#if defined(ANDROID_LIB_STUB) || defined(INTEL_SUPPL_IWL_EXT)
 
 int wpa_driver_set_p2p_noa(void *priv, u8 count, int start, int duration)
 {
@@ -178,7 +178,7 @@ int wpa_driver_set_ap_wps_p2p_ie(void *priv, const struct wpabuf *beacon,
 	return 0;
 }
 
-#endif /* ANDROID_LIB_STUB */
+#endif /* ANDROID_LIB_STUB || INTEL_SUPPL_IWL_EXT */
 #endif /* ANDROID_P2P */
 
 
diff --git a/wpa_supplicant/Android.mk b/wpa_supplicant/Android.mk
index d57446c..4a9ae59 100644
--- a/wpa_supplicant/Android.mk
+++ b/wpa_supplicant/Android.mk
@@ -55,9 +55,15 @@ L_CFLAGS += -Wno-macro-redefined
 # Set Android extended P2P functionality
 L_CFLAGS += -DANDROID_P2P
 
+ifeq ($(BOARD_USING_INTEL_IWL),true)
+L_CFLAGS += -DINTEL_SUPPL_IWL_EXT
+endif
+
 ifeq ($(BOARD_WPA_SUPPLICANT_PRIVATE_LIB),)
+ifndef BOARD_USING_INTEL_IWL
 L_CFLAGS += -DANDROID_LIB_STUB
 endif
+endif
 
 # Disable roaming in wpa_supplicant
 ifdef CONFIG_NO_ROAMING
-- 
2.17.1

